diff --git a/node_modules/dmg-builder/out/dmgUtil.js b/node_modules/dmg-builder/out/dmgUtil.js
index acf576a..c306492 100644
--- a/node_modules/dmg-builder/out/dmgUtil.js
+++ b/node_modules/dmg-builder/out/dmgUtil.js
@@ -99,7 +99,6 @@ async function customizeDmg({ appPath, artifactPath, volumeName, specification,
     const settings = {
         title: path.basename(volumePath),
         icon: await packager.getResource(specification.icon),
-        "badge-icon": await packager.getResource(specification.badgeIcon),
         "icon-size": specification.iconSize,
         "text-size": iconTextSize,
         "compression-level": Number(process.env.ELECTRON_BUILDER_COMPRESSION_LEVEL || "9"),
@@ -114,6 +113,16 @@ async function customizeDmg({ appPath, artifactPath, volumeName, specification,
             // hide_extension: c.hideExtension,
         }))) || [],
     };
+
+    if (specification.badgeIcon) {
+        let badgeIcon = await packager.getResource(specification.badgeIcon);
+        if (badgeIcon.toLowerCase().endsWith(".icon")) {
+            badgeIcon = await packager.generateIcnsFromIcon(badgeIcon)
+        }
+        settings["badge-icon"] = badgeIcon;
+        delete settings["icon"];
+    }
+
     if (specification.backgroundColor != null || specification.background == null) {
         settings["background-color"] = specification.backgroundColor || "#ffffff";
         const window = specification.window;
